### PR TITLE
More fixes

### PR DESCRIPTION
**Pull request by jasongrout**
@blink1073 - here are some updates.

### CODE REVIEW

File: src/docmanager/index.ts [hunk 1]
```diff
@@ -219,11 +219,11 @@ interface IModelFactory {
  * The document manager.
  */
 export
-class DocumentManager {
+class DocumentManager<T extends Widget> {
```
Comment (blink1073) [inline src/docmanager/index.ts]: This won't have the same type as an individual widget factory.

Comment (jasongrout) [inline src/docmanager/index.ts]: The `T` represents the types of widgets the widgetFactory will hand back, right?

Comment (jasongrout) [inline src/docmanager/index.ts]: I'm being dumb here.

Comment (blink1073) [inline src/docmanager/index.ts]: Each widgetFactory has its own T

Comment (jasongrout) [inline src/docmanager/index.ts]: Yep, what I really should have done was make the types in the functions `IWidgetFactory<Widget>`. I fixed it and pushed again.

**Comment by blink1073** [comment]
Thanks!

**Pull request merged**